### ISSUE TITLE
Installation bugs

### ISSUE THREAD

**Issue by diogodaluz** [magento/magento2#2449]
1) I´m getting time out when importing Sample Data. I think the initial verification must verify max_execution_time to prevent it.

2) When I install magento2 without simple data look after install page:

![image](https://cloud.githubusercontent.com/assets/11308015/11309232/c0a60f00-8fa9-11e5-81ec-6e23a131e591.png)

**Comment by thanasisgalanis** [comment]
I have the same issue with a new installation without sample data, but I think that this is a permission problem that not correctly set to files and folders after magento installation.

**Comment by lngla** [comment]
I had both of those issues.

For the max time out I just re-ran the ssh command to install the data several times until it was successful, increasing max_execution_time didn't do much for me.

On the css, images, and java issue: Magento is recommending that permissions for directories be set to 770 and file 660. My server works best by setting permissions to 755 for directories and 644 for files. So after installing and installing sample data, I changed all permissions. I continue to have the issue when adding new products, the image saves to the server under Magento's recommended permissions. It also appears to cache images under the same permissions as well. :/  There is another thread where they are discussing reasons why that doesn't work in certain environments and possible solutions.

**Comment by przmak** [comment]
You sure it's a bug? You sure your virtual host is configured properly and that .htaccess file in the root folder of the project is not missing ?

**Comment by ghost** [comment]
If you have permissions issues, most likely you'll see a 500 on the storefront and, after enabling developer mode, you'll see a stack trace that indicates something like "unable to create directory..."

Your error could be lack of server rewrites: http://devdocs.magento.com/guides/v2.0/install-gde/prereq/apache.html#apache-help-rewrite

**Comment by daim2k5** [comment]
@diogodaluz to you get the time to try out the solution that the other user mentioned?

**Comment by boomtownxcs** [comment]
Hello @diogodaluz, I have ran into the same error which you are running into. To combat this I have needed to change how Magento sets file permissions within the following files: vendor/magento/framework/Filesystem/DriverInterface.php
lib/internal/Cm/Cache/Backend/File.php

This is where Magento is giving the file permissions which do not work in your server configuration. Most of your Magento image cache folders are being set to 770 and images are actually created with 666 permissions. At this time I do not have a better solution and hope that Magento notices that their chose permissions do not work for 90% of the community.
